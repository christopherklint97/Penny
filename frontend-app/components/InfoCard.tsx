import Image from 'next/image'

interface InfoCardProps {
  img: string
  title: string
  body?: string
}

export default function InfoCard({ img, title, body = '' }: InfoCardProps) {
  return (
    <div className="grid justify-center max-w-4xl p-2 mx-auto my-10 text-center align-middle bg-white shadow-md sm:grid-cols-2 rounded-xl">
      <Image
        className="object-cover p-1 mr-auto rounded-lg"
        src={img}
        alt=""
        width="500"
        height="300"
      />
      <div className="p-3 my-auto">
        <h2 className="mx-auto mb-3 text-lg font-semibold">{title}</h2>
        <p className="mx-auto">{body}</p>
      </div>
    </div>
  )
}
