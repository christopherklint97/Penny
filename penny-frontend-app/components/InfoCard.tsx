import {
  Col,
  Row,
  Card,
  CardImg,
  CardBody,
  CardTitle,
  CardText,
} from 'reactstrap'

interface InfoCardProps {
  img: string
  title: string
  body: string
}

export function InfoCard({ img, title, body }: InfoCardProps) {
  return (
    <Row className="my-5">
      <Col className="text-center">
        <Card className="shadow p-3 mb-5 bg-white rounded">
          <Row>
            <Col md="6">
              <CardImg left width="100%" src={img} alt="" />
            </Col>
            <Col className="d-flex align-items-center">
              <CardBody>
                <CardTitle tag="h4" className="yellowtail">
                  {title}
                </CardTitle>
                <CardText>{body}</CardText>
              </CardBody>
            </Col>
          </Row>
        </Card>
      </Col>
    </Row>
  )
}
