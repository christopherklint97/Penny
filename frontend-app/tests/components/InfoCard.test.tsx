import React from 'react'
import { render } from '../testUtils'
import InfoCard from '../../components/InfoCard'

describe('InfoCard component', () => {
  it('renders without crashing', function () {
    render(
      <InfoCard
        title="This is a test"
        img="www.example.com/img"
        body="This is an image"
      />
    )
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <InfoCard
        title="This is a test"
        img="www.example.com/img"
        body="This is an image"
      />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
