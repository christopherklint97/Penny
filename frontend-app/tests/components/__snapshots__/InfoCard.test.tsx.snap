// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid sm:grid-cols-2 shadow-md p-2 bg-white my-10 max-w-4xl mx-auto rounded-xl text-center justify-center align-middle"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block;"
        />
      </div>
      <img
        alt=""
        class="p-1 mr-auto rounded-lg object-cover"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <div
      class="p-3 my-auto"
    >
      <h2
        class="text-lg mx-auto font-semibold mb-3"
      >
        This is a test
      </h2>
      <p
        class="mx-auto"
      >
        This is an image
      </p>
    </div>
  </div>
</DocumentFragment>
`;
